api/envoy/extensions/clusters/dynamic_forward_proxy/v3/cluster.proto

```diff
@@ -53,4 +56,21 @@ message ClusterConfig {
   //   <https://datatracker.ietf.org/doc/html/rfc7540#section-9.1.1>`_
   //
   bool allow_coalesced_connections = 3;
+
+  // Enable strict dns sub cluster instead logic dns.
+  SubClustersConfig sub_clusters_config = 4;
+}
+
+message SubClustersConfig {
```

Comment:
If we aren't going to support full templates, I would structure this as StrictDnsSubClusterConfig, and put this in a oneof above, so that we can potentially add templates later.

Reply:
Added `type` and `lb_policy` for the sub cluster config, they are the necessary template configuration.